build/org.eclipse.cdt.build.gcc.core/src/org/eclipse/cdt/build/gcc/core/ClangToolChain.java

```diff
@@ -28,7 +32,29 @@ public class ClangToolChain extends GCCToolChain {
 
 	public ClangToolChain(IToolChainProvider provider, Path pathToToolChain, String arch,
 			IEnvironmentVariable[] envVars) {
-		super(provider, pathToToolChain, arch, envVars);
+		super(provider, pathToToolChain, arch, addClangEnvVars(envVars));
+	}
+
+	private static IEnvironmentVariable[] addClangEnvVars(IEnvironmentVariable[] envVars) {
+		List<IEnvironmentVariable> envVarsNew = new ArrayList<>(Arrays.asList(envVars));
+		/*
+		 * Set CC and CXX environment variables for clang and clang++. This is equivalent to setting these in the
```

Comment:
Do we also need to set the ASM environment variable?

Reply:
> Do we also need to set the ASM environment variable?
Good question. I do not think so. CMake appears to set ASM automatically to clang when CC is set to clang.

I tested this using the attached project.
[cmakeASM.zip](https://github.com/user-attachments/files/20555699/cmakeASM.zip)

In the screendshot below, it can be seen that the ASM is set to clang.exe.

![cmakeASM helloworld example](https://github.com/user-attachments/assets/174a91f4-0319-4823-b563-b0a28a1f534f)